### ISSUE TITLE
Spec proposal: Remove the global lookup requirements

### ISSUE THREAD

**Issue by jasondlee** [eclipse/microprofile-telemetry#74]
Currently, the spec states this:

```
Calling the OpenTelemetry API directly must work in the same way and yield the same results:
    io.opentelemetry.api.GlobalOpenTelemetry.get()
    io.opentelemetry.api.trace.Span.current()
    io.opentelemetry.api.baggage.Baggage.current()
```
For use cases in application servers such as WildFly, this requirement becomes an issue with multiple deployments due to the way ClassLoaders are shared. Application A creates and sets the global instance, then Application B does the same thing, resulting in:

`IllegalStateException: GlobalOpenTelemetry.set has already been called. GlobalOpenTelemetry.set 
must be called only once before any calls to GlobalOpenTelemetry.get`

From the javadoc for [GlobalOpenTelemetry](https://javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/GlobalOpenTelemetry.html), we see:

`WARNING: Due to the inherent complications around initialization order involving this class and its single global 
instance, we strongly recommend *not* using GlobalOpenTelemetry unless you have a use-case that absolutely 
requires it. Please favor using instances of OpenTelemetry wherever possible.`

Given all of this, we would like to propose dropping the global lookup requirements and leaning solely on CDI injection for these objects.

**Comment by brunobat** [comment]
This should be optional. Single app containers could still allow this.

**Comment by Azquelt** [comment]
Are you proposing dropping just `GlobalOpenTelemetry.get()` or `Span.current()` and `Baggage.current()` as well?

**Comment by jasondlee** [comment]
I'm ok with that and almost suggested that, but it seems that would make portability interesting. Most pressing concern is that it's AT LEAST optional, as that makes implementation for us easier/possible. :)

**Comment by jasondlee** [comment]
> Are you proposing dropping just `GlobalOpenTelemetry.get()` or `Span.current()` and `Baggage.current()` as well?

TBH, I don't know if we'll have the same issue with `Span.current()` or `Baggage.current()`, as the `GlobalOpenTelemetry.get()` is blocking us. Personally, I lean toward excluding those as well, but I don't know if there's a technical reason for that yet.

**Comment by Azquelt** [comment]
`Span.current()` and `Baggage.current()` look like they delegate to `Context.current()` which uses `ContextStorage`. The default `ContextStorage` uses thread local variables which should be fine.

I imagine you would also get other problems if you don't have a working Context.

I would be fine to drop the requirement for `GlobalOpenTelemetry.get()`. I don't see any references to it in the TCK so it looks like it's not being tested anyway. I think the only way to implement it correctly would be to have a special implementation which looks up the correct object for the current application and delegates to it.

I'd prefer to keep `Span.current()` and `Baggage.current()` as I think it's sometimes handy to be able to get those when you're not in a CDI bean and it's possible that there are existing libraries which use them.

**Comment by jasondlee** [comment]
> I'd prefer to keep `Span.current()` and `Baggage.current()` as I think it's sometimes handy to be able to get those when you're not in a CDI bean and it's possible that there are existing libraries which use them.

That all makes sense and sounds fine to me.

**Comment by Emily-Jiang** [comment]
Agreed with the discussion of making the `GloblaOpenTelemetry.get()` optional. The CDI way of looking up either via injection or programmatical lookup via `.current()` should be identical. 
@jasondlee would you like to do a PR today so that we can review/merge in and then restage the final version?

**Comment by jasondlee** [comment]
@Emily-Jiang Working on it now. Spec-only change, correct? I'm not seeing anything in the TCK that seems related.

**Comment by jasondlee** [comment]
I've opened up a PR on this. A little unsure on the exact language, but we can work through that here or on the PR now that there's something to look at, tiny though it may be. :)

**Comment by Emily-Jiang** [comment]
Thank you @jasondlee ! I made a very minor tweak for the sentence. Please take a look to see whether you are okay with it.

**Comment by jasondlee** [comment]
@Emily-Jiang Thanks. Approved and merged into the PR.
